### PR TITLE
Don't reveal secrets in the curl process #1851

### PR DESCRIPTION
**Pull request by sven-dmlr**
also:
- usage infos improved
- alive_check made working without credentials

closes: #1851

### CODE REVIEW

File: sechub-developertools/scripts/pds-api.sh [hunk 1]
```diff
@@ -47,6 +47,11 @@ export PDS_APITOKEN="pds-apitoken"
 EOF
 }
 
+function curl_with_pds_auth {
+  # Don't reveal secrets in the curl process
+  printf "user = $PDS_USERID:$PDS_APITOKEN\n" | curl -K - $CURL_PARAMS "$@"
```
Comment (Jeeppler) [inline sechub-developertools/scripts/pds-api.sh:52]: `-K` is not especially intuitive.

```suggestion
  printf "user = $PDS_USERID:$PDS_APITOKEN\n" | curl --config - $CURL_PARAMS "$@"
```

File: sechub-developertools/scripts/sechub-api.sh [hunk 2]
```diff
@@ -91,6 +91,11 @@ EOF
 #################################
 # Generic helper functions
 
+function curl_with_sechub_auth {
+  # Don't reveal secrets in the curl process
+  printf "user = $SECHUB_USERID:$SECHUB_APITOKEN\n" | curl -K - $CURL_PARAMS "$@"
```
Comment (Jeeppler) [inline sechub-developertools/scripts/sechub-api.sh:96]:
```suggestion
  printf "user = $SECHUB_USERID:$SECHUB_APITOKEN\n" | curl --config - $CURL_PARAMS "$@"
```

**Review by Jeeppler — CHANGES_REQUESTED** [review]
Looks good. I did not even know it is possible.

The only thing I would like to see changed is the parameter `-K` please use the long form `--config` in the scripts.

File: sechub-developertools/scripts/pds-api.sh (hunk 1 above)
Comment (sven-dmlr) [reply sechub-developertools/scripts/pds-api.sh]: implemented

File: sechub-developertools/scripts/sechub-api.sh (hunk 2 above)
Comment (sven-dmlr) [reply sechub-developertools/scripts/sechub-api.sh]: implemented

**Review by Jeeppler — APPROVED** [review]

**Pull request merged**